src/db.rs

```diff
@@ -488,3 +490,51 @@ pub enum DeleteAliasResult {
     /// The alias did not exist.
     Missing,
 }
+
+#[derive(Debug)]
+pub struct Transaction<'conn> {
+    tx: rusqlite::Transaction<'conn>,
+}
+
+impl<'conn> From<rusqlite::Transaction<'conn>> for Transaction<'conn> {
+    fn from(tx: rusqlite::Transaction<'conn>) -> Self {
+        Self { tx }
+    }
+}
+
+impl Transaction<'_> {
+    pub fn commit(mut self) -> rusqlite::Result<()> {
+        self.tx.set_drop_behavior(DropBehavior::Commit);
+        drop(self);
+        Ok(())
+    }
+
+    delegate! {
+        to self.tx {
+            pub fn execute<P>(&self, sql: &str, params: P) ->rusqlite::Result<usize>
+            where
+                P: rusqlite::Params;
+            pub fn last_insert_rowid(&self) -> i64;
+            pub fn pragma_query<F>(&self, schema_name: Option<rusqlite::DatabaseName<'_>>, pragma_name: &str, f: F) -> rusqlite::Result<()>
+            where
+                F: FnMut(&rusqlite::Row<'_>) -> rusqlite::Result<()>;
+            pub fn pragma_query_value<T, F>(&self, schema_name: Option<rusqlite::DatabaseName<'_>>, pragma_name: &str, f: F) -> rusqlite::Result<T>
+            where
+                F: FnOnce(&rusqlite::Row<'_>) -> rusqlite::Result<T>;
+            pub fn prepare(&self, sql: &str) -> rusqlite::Result<rusqlite::Statement<'_>>;
+            pub fn prepare_cached(&self, sql: &str) -> rusqlite::Result<rusqlite::CachedStatement<'_>>;
+        }
+    }
+}
+
+impl Drop for Transaction<'_> {
+    fn drop(&mut self) {
```

Comment:
It just occurred to me that this should probably be `#[inline]`, since in a release build `debug!` should be stripped, which means this function is a no-op.

I mean, almost certainly, in a release build this function call will be entirely removed anyway (there's the joke that the answer to whether or not LLVM will inline a function call is 'yes'). But I guess it doesn't hurt anyway.

Reply:
Done!